notebooks/PGx.ipynb

```diff
@@ -0,0 +1,1180 @@
+    "        }\n",
+    "    }\n",
+    "    \n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "95a73655-acab-4169-9563-b6dd261216f9",
+   "metadata": {},
+   "source": [
+    "A list of definitive sites was extracted from the CYP2C19 Allele Definition Table:\n",
+    "https://files.cpicpgx.org/data/report/current/allele_definition/CYP2C19_allele_definition_table.xlsx"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a64936ee-9ca9-4979-83bd-7abd2b70e517",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "variant_sites = \"g.94761900C>T\tg.94762706A>G\tg.94762712C>T\tg.94762715T>C\tg.94762755T>C\tg.94762760A>C\tg.94762788A>T\tg.94762856A>G\tg.94775106C>T\tg.94775121C>T\tg.94775160G>C\tg.94775185A>G\tg.94775367A>G\tg.94775416T>C\tg.94775423A>C\tg.94775453G>A\tg.94775489G>A\tg.94775507G>A\tg.94780574G>C\tg.94780579G>A\tg.94780653G>A\tg.94781858C>T\tg.94781859G>A\tg.94781944G>A\tg.94781999T>A\tg.94842861G>A\tg.94842866A>G\tg.94842879G>A\tg.94842995G>A\tg.94849995C>T\tg.94852738C>T\tg.94852765C>T\tg.94852785C>G\tg.94852914A>C\".split()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a2c816a2-622f-47da-825b-26c232fa7452",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "site_re = re.compile(r'g.(\\d+)\\w>\\w')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2417d80c-fb02-4ae9-8621-c37f8b98fa44",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "site_positions = [int(site_re.match(x).groups()[0]) for x in variant_sites] "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7546f33b-2c37-47fe-aa2c-a162dbf89052",
+   "metadata": {},
+   "source": [
+    "These sites were used to build VRS [Sequence Location](https://vrs.ga4gh.org/en/stable/terms_and_model.html#sequencelocation) objects."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4ffce75c-d275-49b6-9ffb-08c60e442e63",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "site_seqlocs = list()\n",
+    "for position in site_positions:\n",
+    "    interval = models.SequenceInterval(\n",
+    "        start=models.Number(value=position-1), end=models.Number(value=position))\n",
+    "    seqloc = models.SequenceLocation(\n",
+    "        sequence_id='ga4gh:SQ.ss8r_wB0-b9r44TQTMmVTI92884QvBiB', interval=interval)\n",
+    "    site_seqlocs.append(seqloc.as_dict())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3d77574a-1d2b-467f-a604-ad09fb086d33",
+   "metadata": {},
+   "source": [
+    "This was used to build a domain-specific message structure for CYP2C19 *1:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "70e5856a-06d8-43a5-be5d-0ace27d07221",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "msg = {\n",
+    "    'type': 'StarAllele',\n",
+    "    'variation': h1_allele1.as_dict(),\n",
```

Comment:
`NameError: name 'h1_allele1' is not defined`

Reply:
Great catch! Added 17c1fad to fix.